tests/unit/sstring_test.cc

```diff
@@ -54,6 +54,12 @@ BOOST_AUTO_TEST_CASE(test_add_literal_to_sstring) {
 BOOST_AUTO_TEST_CASE(test_find_sstring) {
     BOOST_REQUIRE_EQUAL(sstring("abcde").find('b'), 1u);
     BOOST_REQUIRE_EQUAL(sstring("babcde").find('b',1), 2u);
+    BOOST_REQUIRE_EQUAL(sstring("").find("", 0), 0u);
+    BOOST_REQUIRE_EQUAL(sstring("").find("", 1), sstring::npos);
+    BOOST_REQUIRE_EQUAL(sstring("abcde").find("", 0), 0u);
+    BOOST_REQUIRE_EQUAL(sstring("abcde").find("", 1), 1u);
+    BOOST_REQUIRE_EQUAL(sstring("abcde").find("", 5), 5u);
+    BOOST_REQUIRE_EQUAL(sstring("abcde").find("", 6), sstring::npos);
```

Comment:
Since the goal of this test (or at least the newly added lines) is to check compatiblity with std::string, this implementation of the test is not convincing enough: you check that find("", 5) should be 5, but who said it should be 5?

What I would have done is to do something like
```c++
auto check_find = [] (const char* s1, const char* s2, int n1, unsigned int n2) {
    BOOST_REQUIRE_EQUAL(sstring(s1).find(s2, n), n2);
    // verify that std::string really has the same behavior as we just tested for sstring
    BOOST_REQUIRE_EQUAL(std::string(s1).find(s2, n), n2);
}
check_find("", "", 0, 0);
check_find("", "", 1, sstring::npos);
...
```

Reply:
test updated. but i found the `npos` of sstring/std::string may not be equal